pkg/apis/gcp/validation/shoot.go

```diff
@@ -60,6 +60,15 @@ func ValidateWorkers(workers []core.Worker, fldPath *field.Path) field.ErrorList
 	return allErrs
 }
 
+// ValidateWorkerAutoScaling checks if the worker.minimum value is greater or equal to the number of worker.zones[]
+// when the worker.maximum value is greater than zero. This check is necessary because autoscaling from 0 is not supported on gcp.
+func ValidateWorkerAutoScaling(worker core.Worker, path string) error {
```

Comment:
Why is this function exported/public?

Reply:
Made it public so that i can use it in the admission package as well:
```
if worker.Minimum < int32(len(worker.Zones)) {
			return fmt.Errorf("%s minimum value must be >= %d if maximum value > 0 (auto scaling to 0 & from 0 is not supported", workersPath.Index(i).Child("minimum").String(), len(worker.Zones))
}
```

was called directly in `validateCreate()` in the admission package and I don't know why it wasn't part of the `ValidateWorkers()` method from the validator package.
Also I don't know why the worker config is generally validated in multiple places, check [this](https://github.com/gardener/gardener-extension-provider-gcp/blob/8b6696e0db54d34993662cfb34bbca4401d60238/pkg/admission/validator/shoot.go#L137) and [this](https://github.com/gardener/gardener-extension-provider-gcp/blob/8b6696e0db54d34993662cfb34bbca4401d60238/pkg/admission/validator/shoot.go#L141-L151) code.